File: AkitaModel/SmartScan/Slot.cs
using Skyware.Lis.AkitaModel.SmartScan.Events;
using System;
using System.Collections.Generic;

namespace Skyware.Lis.AkitaModel.SmartScan;


/// <summary>
/// Represents a slot for uploading files.
/// </summary>
public class Slot
{

    /// <summary>
    /// Base URL for subscribe and unsubscribe
    /// </summary>
    protected readonly string _baseUrl;

    /// <summary>
    /// Default Constructor
    /// </summary>
    public Slot() { }

    /// <summary>
    /// Additional Constructor for creating Slot
    /// </summary>
    /// <param name="baseUrl">Base URL for the internal subscribe/unsubscribe logic</param>
    internal protected Slot(string baseUrl) : this()
    {
        _baseUrl = baseUrl;
    }

    /// <summary>
    /// Id of the slot, server generated.
    /// </summary>
    public string Id { get; set; }

    /// <summary>
    /// Collection of files uploaded in this slot.
    /// </summary>
    public IEnumerable<File> Files { get; set; }

    /// <summary>
    /// Upload mode, <see cref="SlotModes"/>
    /// </summary>
    public SlotModes Mode { get; set; } = SlotModes.PDF;

    /// <summary>
    /// Subscribe to events.
    /// </summary>
    /// <param name="onMobileConnected">Function to be executed when client is 'connected'.</param>
    /// <param name="onFileCreated">Function to be executed when client upload a file.</param>
    public virtual void Subscribe(Action onMobileConnected, Action<FileEvent> onFileCreated) { }

    /// <summary>
    /// Unsubscribe from events.
    /// </summary>
    public virtual void Unsubscirbe() { }

}

File: AkitaMotelTS/src/SmartScan/Slot.ts
/**
 * Skyware.Lis.AkitaModel.SmartScan
 */

import { File } from './File';
import { SlotModes } from './SlotModes';
import { FileEvent } from './Events/FileEvent';

/**
 * Represents a slot for uploading files.
 */
export class Slot {
  /**
   * Base URL for subscribe and unsubscribe
   */
  protected readonly _baseUrl?: string;

  /**
   * Default Constructor
   */
  constructor(baseUrl?: string) {
    if (baseUrl) {
      this._baseUrl = baseUrl;
    }
  }

  /**
   * Id of the slot, server generated.
   */
  id?: string;

  /**
   * Collection of files uploaded in this slot.
   */
  files?: File[];

  /**
   * Upload mode, see SlotModes
   */
  mode: SlotModes = SlotModes.PDF;

  /**
   * Subscribe to events.
   * @param onMobileConnected Function to be executed when client is 'connected'.
   * @param onFileCreated Function to be executed when client upload a file.
   */
  subscribe(onMobileConnected?: () => void, onFileCreated?: (event: FileEvent) => void): void {
    // Implementation would depend on SignalR or WebSocket library
  }

  /**
   * Unsubscribe from events.
   */
  unsubscribe(): void {
    // Implementation would depend on SignalR or WebSocket library
  }
}
